tools/topology/sof-apl-keyword-detect.m4

```diff
@@ -0,0 +1,95 @@
+#
+# Topology for ApolloLake with direct attach digital microphones array for
+# keword detection and triggering use case.
+#
+
+# Include topology builder
+include(`utils.m4')
+include(`dai.m4')
+include(`pipeline.m4')
+
+# Include TLV library
+include(`common/tlv.m4')
+
+# Include Token library
+include(`sof/tokens.m4')
+
+# Include Apollolake DSP configuration
+include(`platform/intel/bxt.m4')
+include(`platform/intel/dmic.m4')
+
+DEBUG_START
+
+#
+# Define the pipelines
+#
+# PCM 0 <-------+- Mux 0 <-- B0 <-- DMIC6 (DMIC01)
+#               |
+# Keyword <-----+
+
+dnl PIPELINE_PCM_ADD(pipeline,
+dnl     pipe id, pcm, max channels, format,
+dnl     frames, deadline, priority, core)
+
+
+# Passthrough capture pipeline 1 on PCM 0 using max 2 channels.
+# Schedule 16 frames per 1000us deadline on core 0 with priority 0
+PIPELINE_PCM_ADD(sof/pipe-mux-capture.m4,
+	1, 0, 2, s32le,
+	16, 1000, 0, 0)
+
+
+# keyword detector pipe
+dnl PIPELINE_ADD(pipeline,
+dnl     pipe id, max channels, format,
+dnl     frames, deadline, priority, core)
+PIPELINE_ADD(sof/pipe-detect.m4, 2, 2, s16le, 16, 1000, 0, 0)
+
+# Connect pipelines together
+SectionGraph."pipe-sof-apl-keyword-detect" {
+        index "0"
+
+        lines [
+		# keyword detect
+                dapm(PIPELINE_SINK_2, PIPELINE_SOURCE_1)
+        ]
+}
+
+#
+# DAIs configuration
+#
+
+dnl DAI_ADD(pipeline,
+dnl     pipe id, dai type, dai_index, dai_be,
+dnl     buffer, periods, format,
+dnl     frames, deadline, priority, core)
+
+
+# capture DAI is DMIC 0 using 1000 periods (64kB buffer for 2 secs of audio)
+# Buffers use s16le format, with 16 frame per 1000us on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-capture.m4,
+	1, DMIC, 0, NoCodec-6,
```

Comment:
is NoCodec-6 valid here? this doesn't seem right

Reply:
@singalsu can you comment on DMIC settings as these are the values you created last week.